homeassistant/components/nextcloud/entity.py

```diff
@@ -21,3 +23,14 @@ def __init__(self, coordinator: NextcloudDataUpdateCoordinator, item: str) -> No
     def unique_id(self) -> str:
         """Return the unique ID for this sensor."""
         return f"{self.coordinator.url}#{self.item}"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device specific attributes."""
+        return DeviceInfo(
+            name=self.coordinator.url,
+            identifiers={(DOMAIN, self.coordinator.url)},
+            model="Nextcloud",
+            sw_version=self.coordinator.data.get("nextcloud_system_version"),
+            configuration_url=self.coordinator.url,
+        )
```

Comment:
Maybe move `device_info` to another PR.
I am not sure that the url is a valid identifier and I think it should have it's own review process.